### ISSUE TITLE
When having both named subscription and queries, query is sent over websockets

### ISSUE THREAD

**Issue by codepunkt** [graphql/graphiql#606]
When having both named subscription and queries like this

![queryoverws](https://user-images.githubusercontent.com/603683/30850233-2170f51a-a2a5-11e7-873f-4eb2679abc1c.PNG)

and selecting to send query a over the wire, it is send via websockets (without operationName, see #605).

I suppose this is a bug?

**Comment by vunguyenhung** [comment]
The same here. Graphiql performs the same behavior with named Mutation and Subscription combination. None of them work unless I split them into separate tab/window.

**Comment by sjparsons** [comment]
**tl;dr** I have verified that `operationName` is not sent went using subscriptions, but the root cause is not in the GraphiQL package itself.

---

**Reproducing**

I temporarily hacked in support for GraphQL over websocket to the test GraphQL server in the `graphiql` package to help diagnose other subscriptions/ws issues.

I found that if I do a query like this...

```gql
query a {
  test {
    id
  }
}

subscription b {
  subscribeToTest(id: "my") {
    test {
      id
    }
  }
}
```

... and run operation `a` or `b`, you will be able to see successful interaction over the websocket, but you'll see if you drill into the network dev tools messages like this being sent, with `operationName` missing:

```json
{
    "query": "\nquery a {\n test {\n id\n }\n}\n\n# query b {\n# test {\n# id\n# }\n# }\n\n\nsubscription b {\n subscribeToTest(id: \"my\") {\n test {\n id\n }\n }\n}\n\n",
    "variables": null,
    "type": "subscription_start",
    "id": 0
}
```

You can wrap the  GraphiQL `fetcher` with a function including a `debug`/`console.log` and verify that GraphiQL's call to the fetcher includes the `operationName`. 

**Root cause**

As I noted above, GraphiQL is correctly handing the `operationName` to the fetcher, but the common fetcher to use with GraphiQL for subscriptions,  https://github.com/apollographql/GraphiQL-Subscriptions-Fetcher does not pass the operationName along. 

https://github.com/apollographql/GraphiQL-Subscriptions-Fetcher/blob/master/src/fetcher.ts#L34

Also, rather importantly, that repo has been archived!

**Possible next steps:**

1) Close this issue since the heart of the matter is not GraphiQL itself. 

However, I'm not 100% satisfied with that answer. No one is likely to fix this issue or provide a workaround as the source of the issue (one that we reference in our `test` that used in `yarn dev`) is archived. So maybe we could:

2. Provide an example GraphiQL fetcher for subscriptions that works. 
   - Migrate https://github.com/apollographql/GraphiQL-Subscriptions-Fetcher/blob/master/src/fetcher.ts into this repo (maybe as a different package?)
   - Update our test GraphiQL example (the one used with `yarn dev`) to use that fetcher.

To help support the approach of doing (2) I did some quick testing and verified that we can pass the `operationName` along if we were able to change the fetcher.

**Comment by acao** [comment]
Thanks so much for this very well constructed & researched bug report.

We should indeed figure out a more robust solution for subscribing to wss
for graphiql. I can ask Uri if this code is still alive elsewhere, mayhaps?
And yes, lastly creating our own client just for graphiql makes sense,
eventually as a plugin (as will all ws/wss subscriptions support)

**Comment by sjparsons** [comment]
Also, so we don't lose track, this issue is related to the same underlying issue: 

https://github.com/graphql/graphiql/issues/605

**Comment by acao** [comment]
thanks @sjparsons! hopefully we can get to this soon after resolving some higher priority bugs

**Comment by acao** [comment]
this should be fixed here:
https://github.com/graphql/graphiql/pull/664
